### PR TITLE
Bug - 5031 - Require skill details application

### LINKED ISSUE #5031: Validation of skills during application should check for justifications.
# ⚙️ Summary

We are getting users through the application process that have claimed a skill, but have not filled out a justification.

This was an unintended consequence of removing skill justifications to skills when added on the profile.

The current skill validation used on the application checks only whether a skill is on the profile, but what we really need is to know wether or not a user has justified their skill on the experience.

We should expand the validation of skills to include those skill justifications and create appropriate error messages for them.

Any questions please ask: @Jerryescandon 

### [Visit the Figma file]()

[PURPOSE AND DESCRIPTION]

## 💾 Files and resources

image

## ✅ Design checklist

- [ ] Component library elements have been used where appropriate
- [ ] Component library elements have been updated where changes where required
- [ ] Newly added interface elements have been added to the component library for reuse
- [ ] Newly added interface elements have been added to Github Discussions using the component issue template
- [ ] The interface is responsive or provides responsive context
- [ ] The interface provides both light and dark mode designs
- [ ] Complex interactions have been prototypes in a way that showcases every step in the flow
- [ ] Empty/null states have been accounted for
- [ ] The design has been reviewed by at least one other team member
- [ ] The design has been presented to the design team
- [ ] The design has been approved by stakeholders

## ✅ Accessibility checklist

**Please refer to [the comprehensive list](https://www.figma.com/file/rRMvm5nq4AMet29PW7bN2p/A11y-annotations?node-id=0%3A1) for more details.**

- [ ] Links have meaningful label text
- [ ] Navigation matches previously established patterns
- [ ] Headings are identified using annotations and follow a logical hierarchy
- [ ] Layout is linear and consistent with established patterns
- [ ] New interactive elements have unique focus states that match the established pattern
- [ ] Tab order has been identified using annotations
- [ ] Relevant images and icons have appropriate `alt` text
- [ ] Use of color meets AA WCAG contrast requirements
- [ ] States or information are not identified by color alone
- [ ] New interactive elements have appropriately sized touch targets on touch devices

## 📸 Screenshot

[SCREENSHOT]

# ✅ Requirements

- [ ] [REQUIREMENT]

## 🛠️ [REQUIREMENT NAME]

[REQUIREMENT DESCRIPTION]

[REQUIREMENT SCREENSHOT]

### [REQUIREMENT NAME]

[REQUIREMENT DESCRIPTION]

[REQUIREMENT SCREENSHOT]

# 🤔 Outstanding concerns

- [CONCERN]

# 📝 Related issues

**Please link all design and development sub-tasks related to this issue here for tracking purposes.**

- [ISSUE #]

### PR DESCRIPTION
**Pull request by esizer**
## 👋 Introduction

This updated missing skills to differentiate between skills that are entirely missing from a profile and those that are only missing details.

> **Note**
> Part of the acceptance criteria was to prevent submission when missing details. However, this was fixed with #4957 but has yet to be deployed so that is not included in this PR.

## ✅ TODO

- [x] Confirm AC with @Jerryescandon 
- [x] Send off for translations
- [x] Add translations

## 🧪 Testing

1. Build talent search `npm run production --workspace=talentsearch`
2. Create or navigate to an existing application
3. Check missing skills
4. Add a missing skill **_from your profile, not the application page_** (we want one that is added but had no details)
5. Refresh the application reivew page
6. Confirm the new skill appears under the label "These required skills are missing information:"

## 🤖 Robot Stuff

Resolves #5031

### CODE REVIEW

**Review by petertgiles — CHANGES_REQUESTED** [review]
This works great!  I'm finding it pretty hard to follow though.  Would you mind doing a pass on it to improve commenting, parameter naming, and varible naming?

File: frontend/common/src/helpers/skillUtils.ts [hunk 1]
```diff
@@ -143,4 +143,24 @@ export const getMissingSkills = (required: Skill[], added?: Skill[]) => {
       });
 };
 
+export const differentiateMissingSkills = (
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/common/src/helpers/skillUtils.ts:158]: Added some documentation in [improve missing skills documentation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5105/commits/59cfe21c4defd68187c10116a660ec0c5f299041) to hopefully clarify what is going on here.

[the remark this answers is not in the document]
Comment (petertgiles) [reply frontend/common/src/helpers/skillUtils.ts:158]: Much better, thanks!

**Review by petertgiles — APPROVED** [review]
